demo/main.js

```diff
@@ -259,6 +278,18 @@ class Demo extends Component {
     const result = await fetch("http://localhost:9000");
     return result.json();
   }
+
+  getImageSize(path) {
```

Comment:
I think it's a naming issue, it should be `figure.size` and `image.resolution`, and it would be clearer. 

On precomputing it or not, I don't really care. The ImageProvider seems to already have a `getImageSize` method, so we could probably just use that if the information isn't needed in a plugin.